src/components/CalendarPicker/index.js

```diff
@@ -0,0 +1,169 @@
+import _ from 'underscore';
+import React from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import moment from 'moment';
+import Text from '../Text';
+import ArrowIcon from './ArrowIcon';
+import styles from '../../styles/styles';
+// eslint-disable-next-line rulesdir/prefer-import-module-contents
+import {calendarPickerPropType, defaultCalendarPickerPropType} from './calendarPickerPropTypes';
+import generateMonthMatrix from './generateMonthMatrix';
+import withLocalize from '../withLocalize';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+
+class CalendarPicker extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.monthNames = moment.localeData(props.preferredLocale).months();
+        this.daysOfWeek = moment.localeData(props.preferredLocale).weekdays();
+
+        let currentDateView = props.value;
+        if (props.maxDate < currentDateView) {
+            currentDateView = props.maxDate;
+        } else if (props.minDate > currentDateView) {
+            currentDateView = props.minDate;
+        }
+
+        this.state = {
+            currentDateView,
+        };
+
+        this.onPrevMonthPressed = this.onPrevMonthPressed.bind(this);
+        this.onNextMonthPressed = this.onNextMonthPressed.bind(this);
+        this.onYearPickerPressed = this.onYearPickerPressed.bind(this);
+        this.onDayPressed = this.onDayPressed.bind(this);
+    }
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    componentDidMount() {
+        if (this.props.minDate > this.props.maxDate) {
+            throw new Error('Minimum date cannot be greater than the maximum date.');
+        }
+    }
```

Comment:
I don't think we should disable this, instead we should do what the rule says and return early
```suggestion
    // eslint-disable-next-line rulesdir/prefer-early-return
    componentDidMount() {
        if (this.props.minDate <= this.props.maxDate) {
            return;
        }
        throw new Error('Minimum date cannot be greater than the maximum date.');
    }
```